### PR TITLE
[Feature Store] Fix iso conversion on parquet source

### PR DESCRIPTION
**Pull request by benbd86**
Handle issues with conversion on datetime received from java in iso format.
Try conversion of both java and python iso formats.

[ML-1133](https://jira.iguazeng.com/browse/ML-1133)

### CODE REVIEW

File: mlrun/datastore/sources.py [hunk 1]
```diff
@@ -239,6 +233,39 @@ def __init__(
...
+        else:
+            self._end_time = end_time
+
+    @staticmethod
+    def _convert_to_datetime(time: str):
+        try:
+            return datetime.fromisoformat(time)
+        except ValueError as error:
+            # check if java's iso format if so remove the Z from the end and try again
+            if time.endswith("Z"):
```
Comment (gtopper) [inline mlrun/datastore/sources.py:264]: I think it makes more sense to first drop a trailing Z if necessary and then parse. Then there is no error handling to consider.

File: mlrun/datastore/sources.py [hunk 2]
```diff
@@ -239,6 +233,39 @@ def __init__(
...
+            self._end_time = self._convert_to_datetime(end_time)
+        else:
+            self._end_time = end_time
+
+    @staticmethod
+    def _convert_to_datetime(time: str):
+        try:
+            return datetime.fromisoformat(time)
+        except ValueError as error:
+            # check if java's iso format if so remove the Z from the end and try again
```
Comment (gtopper) [inline mlrun/datastore/sources.py:263]: There is no such thing as "java's iso format" 😅

File: mlrun/datastore/sources.py (hunk 1 above)
Comment (gtopper) [reply mlrun/datastore/sources.py]: Also, shouldn't we transform `Z` to `+00:00` rather than just erase it?

**Review by gtopper — APPROVED** [review]

File: tests/system/feature_store/test_feature_store.py [hunk 3]
```diff
@@ -2489,6 +2488,61 @@ def test_get_offline_for_two_feature_set_with_same_column_name(self):
         for key in res.to_dataframe().to_dict().keys():
             assert key in expected
 
+    @pytest.mark.parametrize(
```
Comment (Hedingber) [inline tests/system/feature_store/test_feature_store.py:2491]: I don't see a reason for this test to be a system test (expensive/slow) and not a unit test (cheap/fast)

File: tests/system/feature_store/test_feature_store.py [hunk 4]
```diff
@@ -2489,6 +2488,61 @@ def test_get_offline_for_two_feature_set_with_same_column_name(self):
...
+            ("2021-05-25T10:30:29.592+00:00", False, True, 0),
+            ("2021-05-25T10:30:29.592+00:00", True, True, 0),
+            ("2021-05-25T10:30:29.592+02:00", False, True, timedelta(hours=2)),
+            ("2021-05-25T10:30:29.592+02:00", True, True, timedelta(hours=2)),
+        ],
+    )
+    def test_parquet_source_with_iso_start_or_end_time(
+        self, time_for_source, is_through_init, should_succeed, time_delta
+    ):
+        def actual_test(time_for_source, is_through_init, time_delta):
```
Comment (Hedingber) [inline tests/system/feature_store/test_feature_store.py:2514]: If there's no more specific name then just the name of the test function simply go with (prefix underscore) `_test_parquet_source_with_iso_start_or_end_time`

**Review by Hedingber — CHANGES_REQUESTED** [review]

**Review by Hedingber — APPROVED** [review]

**Pull request merged**